explorer/explorer.go

```diff
@@ -109,27 +107,51 @@ func GetTicketPoolData(interval dbtypes.ChartGrouping) (barGraphs []*dbtypes.Poo
 	return
 }
 
-// cleanUpTicketPoolData provides a thread-safe way to clean up/invalidate the ticketspool
-// cache data. This function should be called when a new block has been added.
-func cleanUpTicketPoolData() {
-	ticketPoolGraphsCache.Lock()
-	defer ticketPoolGraphsCache.Unlock()
+// CleanUpTicketPoolData provides a thread-safe way to clean up/invalidate the
+// ticketpool cache data. It resets the ticketpool cache data and sets the
+// current block height. Check for invalid data every cleanUpTickerInterval.
+func (exp *explorerUI) CleanUpTicketPoolData(quit chan int) {
+	ticker := time.NewTicker(time.Duration(cleanUpTickerInterval) * time.Second)
+	for {
+		select {
+		case <-ticker.C:
+			height := exp.Height()
+			// if data is updated do not proceed
+			if ticketPoolGraphsCache.Height == height {
+				continue
+			}
+
+			ticketPoolGraphsCache.Lock()
+			defer ticketPoolGraphsCache.Unlock()
 
-	ticketPoolGraphsCache = &ticketPoolDataCache{
-		BarGraphsCache:  make(map[dbtypes.ChartGrouping][]*dbtypes.PoolTicketsData),
-		DonutGraphCache: make(map[dbtypes.ChartGrouping]*dbtypes.PoolTicketsData),
+			ticketPoolGraphsCache = &ticketPoolDataCache{
+				Height:          height,
+				BarGraphsCache:  make(map[dbtypes.ChartGrouping][]*dbtypes.PoolTicketsData),
+				DonutGraphCache: make(map[dbtypes.ChartGrouping]*dbtypes.PoolTicketsData),
+			}
+		case <-quit:
+			log.Debug("Stopping the ticketpool cleanUp")
+			ticker.Stop()
+		default:
+		}
 	}
 }
 
 // UpdateTicketPoolData updates the ticket pool cache with the latest data fetched.
-// This is a thread-safe way to update ticket pool cache data.
+// This is a thread-safe way to update ticket pool cache data. TryLock helps avoid
+// stacking calls to update the cache.
 func UpdateTicketPoolData(interval dbtypes.ChartGrouping, barGraphs []*dbtypes.PoolTicketsData,
 	donutcharts *dbtypes.PoolTicketsData) {
+	isLocking := ticketPoolGraphsCache.updating.TryLock()
```

Comment:
The "is already updating?" check needs to be where the DB query is launched since that's what we want to prevent stacking up.

Reply:
ok. Noted